crates/uv-build-backend/src/lib.rs

```diff
@@ -77,6 +77,8 @@ pub enum Error {
 /// error case).
 trait DirectoryWriter {
```

Comment:
It strikes me that this trait seems like a prime candidate for virtual dispatch. But I think I see below that it is using monomorphization. The `close` method looks like that might be a bugger, but maybe there is a way to make it work. (Or give up taking `self` by value, regrettably.)